### ISSUE TITLE
Cannot use the lower function in a criteria

### ISSUE THREAD

**Issue by fmo-dev** [squashql/squashql#320]
In a use case i need to do the equivalent of 
```sql
LOWER( product.productName ) LIKE  '%term%'
```

Sadly, this seems not to be possible  because using the `lower` function in a criteria like this 
```ts
criterion(lower(product.productName), like(`%${term}%`))
``` 

lead to a crash of the application with this error

![image](https://github.com/squashql/squashql/assets/64314882/7d73502b-88a4-45b7-85e6-ba85ccfef1a4)

It seems that the problem come from the missing case for `_FunctionField` in the function  `computeFieldDependencies `

**Comment by paulbares** [comment]
thank you for reporting this issue, I will fix it asap.

**Comment by paulbares** [comment]
Fixed by #322
